azure-iot-hub-devicesdk/tests/test_mqtt_transport.py

```diff
@@ -17,7 +20,61 @@
 fake_device_id = "MyPensieve"
 fake_event = "Wingardian Leviosa"
 fake_event_2 = fake_event + " again!"
-fake_topic = "devices/" + fake_device_id + "/messages/events/"
+
+message_id = "spell-1234"
+user_id = "HG749134"
+
+
+def create_fake_message(telemtry_str):
+    msg = Message(telemtry_str)
+    msg.message_id = "spell-1234"
+    msg.user_id = "HG749134"
+    msg.custom_properties["name"] = "hermoinee granger"
+    msg.custom_properties["school"] = "hogwarts"
+    msg.custom_properties["birthstatus"] = "muggle"
+    msg.custom_properties["profession"] = "witch"
+    return msg
+
+
+def create_encoded_fake_topic(fake_message):
```

Comment:
>create_encoded_fake_topic [](start = 4, length = 25)

that looks like a duplication of the code we have in the transport. meaning if we have a bug there, we have it here.

I'd much rather see code that deconstructs the topic (splits, uri decoding) based on what we expect and check that we can parse the properties the way the service would. sort of "reverse" our encoding algorithm and verify we fall back on our feet, if you will

Reply:
I am not sure what you mean by the reverse in this context since I need the topic to be created for an assertion. This was just a helper. I have created the topic without the code now.

---
In reply to: [240414779](https://github.com/Azure/azure-iot-sdk-python-preview/pull/34#discussion_r240414779) [](ancestors = 240414779)